blockscout-ens/bens-logic/src/subgraphs_reader/sql/address_names.rs

```diff
@@ -0,0 +1,87 @@
+use crate::subgraphs_reader::SubgraphReadError;
+use sqlx::{postgres::PgPool, Executor};
+use tracing::instrument;
+
+#[instrument(
+    name = "create_address_names_view",
+    skip(pool),
+    err(level = "error"),
+    level = "info"
+)]
+pub async fn create_address_names_view(
+    pool: &PgPool,
+    schema: &str,
+) -> Result<(), SubgraphReadError> {
+    let mut tx = pool.begin().await?;
+
+    tx.execute(sqlx::query(&format!(
+        r#"
+        CREATE MATERIALIZED VIEW IF NOT EXISTS {schema}.address_names AS
+        SELECT DISTINCT ON (resolved_address)
+            id,
+            name AS domain_name,
+            resolved_address
+        from {schema}.domain
+        where
+            resolved_address IS NOT NULL
+            AND label_name IS NOT NULL
+            AND name NOT LIKE '%[%'
+            AND (
+                expiry_date is null
+                OR to_timestamp(expiry_date) > now()
+            )
+            AND block_range @> 2147483647
```

Comment:
It is better to reuse `DOMAIN_NOT_EXPIRED_WHERE_CLAUSE` and `DOMAIN_DEFAULT_WHERE_CLAUSE` constants to make it consistent between the view and the actual  query